### ISSUE TITLE
Trigger operation does not take parameter

### ISSUE THREAD

**Issue by MartinOrell** [secure-dev-ops/code-realtime#14]
It seems like adding a parameter to a trigger does not work.

I have a class with a defined trigger with a parameter which looks the same as the one provided in the documentation.
I try to call the trigger with one parameter.
I get an error with a note: candidate expects 0 arguments, 1 provided

The code executes when I remove the parameter in the function call.
The variable inside the transition is undefined.

My code:
```
class DataObject {
    [[rt::header_preface]]
    `
        #include <iostream>
    `

    /* Trigger Operations */
    trigger initialize(`int` data);
    /* State Machine */
    statemachine {
        state Initial, Initialized;
        initial -> Initial;
        init: Initial -> Initialized on initialize(`int`)
        `
            // Initialized
            //int i = data; //Produces error: 'data' was not declared in this scope
            std::cout << "Initialized" << std::endl;
        `;
    };
};

capsule Main{
    [[rt::header_preface]]
    `
        #include "DataObject.h"
        #include <string>
    `
    [[rt::decl]]
    `
    private:
        DataObject* _dataObject;
    `

    statemachine {
        state Hello;
        initial -> Hello
        `
            _dataObject = new DataObject();
            int i = 1;
            _dataObject->initialize(i);
            // _dataObject->initialize(); //This works without arguments
            context()->abort();
        `;
    };
};
```

**Comment by alexander-str** [comment]
@MartinOrell This appears to be a bug in the code generator for trigger operations with parameters. The dev team will look into it and update you.

**Comment by mattias-mohlin** [comment]
This has now been fixed and will be available in version 2.0.0 to be released around September 18.
